File: receptionHubBackend/Controllers/Dtos/SobaDto.cs
using receptionHubBackend.Models.Enums;

namespace receptionHubBackend.Controllers.Dtos;

public class SobaDto
{
    public int IdSobe { get; set; }
    public string BrojSobe { get; set; } = null!;
    public TipSobe TipSobe { get; set; }
    public int MaksimalnoGostiju { get; set; }
    public int BrojKreveta { get; set; }
    public int? BrojBracnihKreveta { get; set; }
    public int? BrojOdvojenihKreveta { get; set; }
    public bool ImaDodatniKrevet { get; set; }
    public decimal CijenaPoNociBAM { get; set; }
    public string? Opis { get; set; }
    public string? KratkiOpis { get; set; }

    // Oprema
    public bool ImaTv { get; set; }
    public bool ImaKlimu { get; set; }
    public bool ImaMiniBar { get; set; }
    public bool ImaPogledNaGrad { get; set; }
    public bool ImaWiFi { get; set; }
    public bool ImaRadniSto { get; set; }
    public bool ImaFen { get; set; }
    public bool ImaPeglu { get; set; }
    public bool ImaKupatilo { get; set; }
    public bool ImaTus { get; set; }

    // Status
    public StatusSobe Status { get; set; }
    public DateTime? PlaniranoOslobadjanje { get; set; }

    // Slike i napomene
    public string? GlavnaSlika { get; set; }
    public string? Napomena { get; set; }

    // Metadata
    public DateTime VrijemeKreiranja { get; set; }
    public int? KreiraoRecepcionerId { get; set; }
    public DateTime? VrijemeAzuriranja { get; set; }

    // Dodatno - broj aktivnih rezervacija (opciono)
    public int BrojAktivnihRezervacija { get; set; }
}

File: receptionHubFrontend/src/types/dtos/soba.dto.ts
export enum TipSobe {
  Standard = 0,
  Superior = 1,
  Deluxe = 2,
  Apartman = 3,
  Studios = 4
}

export enum StatusSobe {
  Slobodna = 0,
  Zauzeta = 1,
  Ciscenje = 2,
  Rezervisana = 3,
  VanFunkcije = 4
}

export interface SobaDto {
  idSobe: number;
  brojSobe: string;
  tipSobe: TipSobe;
  maksimalnoGostiju: number;
  brojKreveta: number;
  brojBracnihKreveta?: number;
  brojOdvojenihKreveta?: number;
  imaDodatniKrevet: boolean;
  cijenaPoNociBAM: number;
  opis?: string;
  kratkiOpis?: string;
  
  // Oprema
  imaTv: boolean;
  imaKlimu: boolean;
  imaMiniBar: boolean;
  imaPogledNaGrad: boolean;
  imaWiFi: boolean;
  imaRadniSto: boolean;
  imaFen: boolean;
  imaPeglu: boolean;
  imaKupatilo: boolean;
  imaTus: boolean;
  
  // Status
  status: StatusSobe;
  planiranoOslobadjanje?: string; // ISO date string
  
  // Slike i napomene
  glavnaSlika?: string;
  napomena?: string;
  
  // Metadata
  vrijemeKreiranja: string;
  kreiraoRecepcionerId?: number;
  vrijemeAzuriranja?: string;
  
  // Dodatno
  brojAktivnihRezervacija: number;
}

export interface KreirajSobuDto {
  brojSobe: string;
  tipSobe: TipSobe;
  maksimalnoGostiju: number;
  brojKreveta: number;
  brojBracnihKreveta?: number;
  brojOdvojenihKreveta?: number;
  imaDodatniKrevet?: boolean;
  cijenaPoNociBAM: number;
  opis?: string;
  kratkiOpis?: string;
  
  imaTv?: boolean;
  imaKlimu?: boolean;
  imaMiniBar?: boolean;
  imaPogledNaGrad?: boolean;
  imaWiFi?: boolean;
  imaRadniSto?: boolean;
  imaFen?: boolean;
  imaPeglu?: boolean;
  imaKupatilo?: boolean;
  imaTus?: boolean;
  
  glavnaSlika?: string;
  napomena?: string;
}

export interface AzurirajSobuDto {
  brojSobe?: string;
  tipSobe?: TipSobe;
  maksimalnoGostiju?: number;
  brojKreveta?: number;
  brojBracnihKreveta?: number;
  brojOdvojenihKreveta?: number;
  imaDodatniKrevet?: boolean;
  cijenaPoNociBAM?: number;
  opis?: string;
  kratkiOpis?: string;
  
  imaTv?: boolean;
  imaKlimu?: boolean;
  imaMiniBar?: boolean;
  imaPogledNaGrad?: boolean;
  imaWiFi?: boolean;
  imaRadniSto?: boolean;
  imaFen?: boolean;
  imaPeglu?: boolean;
  imaKupatilo?: boolean;
  imaTus?: boolean;
  
  status?: StatusSobe;
  planiranoOslobadjanje?: string;
  
  glavnaSlika?: string;
  napomena?: string;
}

export interface StatusSobeDto {
  status: StatusSobe;
  planiranoOslobadjanje?: string;
}